src/ui/ui_i18n.js

```diff
@@ -0,0 +1,126 @@
+const I18n =  class {
+  constructor(translations) {
+    this.translations = translations;
+  }
+
+  translate(key) {
+    if (!this.translations.hasOwnProperty(key)) {
+      return null;
+    }
+    return this.translations[key];
+  }
+};
+
+const getLocaleTranslations = async function (locale, server) {
```

Comment:
I think most of this complexity belongs in the core i18n module. Here's what I'm thinking:

We should extend i18n's `getRegisteredLocaleTranslations` to accept one or more BCP 47 language tags. The sole responsibility of the code under `src/ui` is to parse the accept-language header, construct a prioritized list of language tags, and pass it to `getRegisteredLocaleTranslations`. The i18n module should contain all the logic for finding the best match for a given prioritized list of language tags. In fact maybe we make it a separate method, `getBestLocaleTranslations` or something.